data_collection/gazette/spiders/sp_sorocaba.py

```diff
@@ -0,0 +1,100 @@
+import re
+from datetime import date, datetime
+
+from scrapy import Request
+from dateparser import parse
+from dateutil.rrule import MONTHLY, rrule
+
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class SpSorocabaSpider(BaseGazetteSpider):
+    name = "sp_sorocaba"
+    TERRITORY_ID = "3552205"
+
+    start_date = date(2005, 1, 7)
+
+    gazettes_list_url = "http://noticias.sorocaba.sp.gov.br/jornal/?ano={}"
+
+    MonthDict = {
+        1: "Janeiro",
+        2: "Fevereiro",
+        3: "Marco",
+        4: "Abril",
+        5: "Maio",
+        6: "Junho",
+        7: "Julho",
+        8: "Agosto",
+        9: "Setembro",
+        10: "Outubro",
+        11: "Novembro",
+        12: "Dezembro",
+    }
+
+    def start_requests(self):
+        end_date = date.today()
+
+        periods_of_interest = [
+            (date.year, date.month)
+            for date in rrule(freq=MONTHLY, dtstart=self.start_date, until=end_date)
+        ]
+        global year
+        for year, month_int in periods_of_interest:
+            month_text = self.MonthDict[month_int]
+            period = f"{ year }+{ month_text }"
+            url = self.gazettes_list_url.format(period)
+            yield Request(url, self.parse)
+
+    def parse(self, response):
+        for link in response.css("div.list-group-flush.w-100 a::attr(href)"):
+            url = self.get_pdf_url(response, link)
+            if not url:
+                continue
+
+            [gazette_date, gazette_edition_number] = self.get_date_and_edition_number(
+                link
+            )
+
+            yield Gazette(
+                date=gazette_date,
+                edition_number=gazette_edition_number,
+                file_urls=(url,),
+                is_extra_edition=False,
+                territory_id=self.TERRITORY_ID,
+                power="executive_legislature",
+                scraped_at=datetime.utcnow(),
+            )
+
+        next_page = response.css("li.next.page-item a::attr(href)").extract()
+		
+		if next_page:

```

Comment:
**issue (blocking)**: As already mentioned by @jvanz, you are mixing tabs and spaces in your code. Before submitting a PR it is always a good practice to check if there is no changes to be commited (using `git status`), and then run your code. Sometimes we finish the code, it runs as we want in our machine, but we forget to commit all the changes and sometimes the code that are pushed is not the latest version.